generate_user_guides.sh

```diff
@@ -1,6 +1,14 @@
 #!/bin/bash
 set -v
 
+if [ gsed -v ]; then
+    gsed=gsed
+else
+    # Hopefully installed sed is the gnu one.
+    # note: aliases work poorly with xargs.
+    gsed=sed
+fi
```

Comment:
using `gsed` is not trivial on linux actually, so this "hack" allows for better cross platform tooling.